lib/repository/online_graphql_repository/plan_graphql_model.dart

```diff
@@ -1,12 +1,41 @@
 import 'package:flutter/material.dart';
-import 'package:trufi_core/models/utils_models.dart';
 import 'package:trufi_core/trufi_models.dart';
 
-enum _Mode { walk, rail, bus }
+enum TransportMode { walk, rail, bus, transit, car }
+
+TransportMode _getEnumMode(String enumS) {
+  switch (enumS) {
```

Comment:
Instead of a switch, I like the idea of a map, what do you think?
https://dartpad.dartlang.org/c4001d907d6a420cafb2bc2c2507f72c

Reply:
It's good to use a map in the extensionEnum class if we do _getEnumMode a map, We need to control the response null when we need to use the map we need to put "??" for control value null.

can the map help me to control default values how the switch?

Reply:
enum TransportMode { walk, rail, bus, transit, car }

const transportModeMap= {
  'WALK': TransportMode.walk,
  'BUS': TransportMode.bus,
  'CAR': TransportMode.car,
  'LIGHT RAIL': TransportMode.rail,
  'TRANSIT': TransportMode.transit,
};

TransportMode _getTransportMode(String enumText) 
  => transportModeMap[enumText] ?? TransportMode.walk;

extension TransportModeExtension on TransportMode {
  static const names = {
    TransportMode.walk: 'WALK',
    TransportMode.bus: 'BUS',
    TransportMode.car: 'CAR',
    TransportMode.rail: 'LIGHT RAIL',
    TransportMode.transit: 'TRANSIT',
  };
  String get name => names[this] ?? '';
}